File: src/DVLD.Server/DVLD.Application/DTOs/PagedList.cs
namespace DVLD.Application.DTOs;

public sealed class PagedList<T>(
    IReadOnlyList<T> items,
    int totalCount,
    int pageNumber,
    int pageSize)
{
    public IReadOnlyList<T> Items { get; } = items;
    public int PageNumber { get; } = pageNumber;
    public int PageSize { get; } = pageSize;
    public int TotalCount { get; } = totalCount;
    public int TotalPages { get; } = (int)Math.Ceiling(
        totalCount / (double)pageSize);

    public bool HasNextPage => PageNumber < TotalPages;
    public bool HasPreviousPage => PageNumber > 1;
}

File: src/DVLD.Client/src/app/shared/models/paged-list.ts
export class PagedList<T> {
  constructor(
    public readonly items: ReadonlyArray<T>,
    public readonly totalCount: number,
    public readonly pageNumber: number,
    public readonly pageSize: number,
  ) {}

  public get totalPages(): number {
    return Math.ceil(this.totalCount / this.pageSize);
  }

  public get hasNextPage(): boolean {
    return this.pageNumber < this.totalPages;
  }

  public get hasPreviousPage(): boolean {
    return this.pageNumber > 1;
  }
}
